MapsAndPlacesDemo/MapsAndPlacesDemo/LocationImageGenerator.swift

```diff
@@ -0,0 +1,99 @@
+    ///   - select: Indicates if we want the image on the image view; if this is false, tapped should be true.
+    ///   - tapped: Indicates if we want the image on the GMSMarker; if this is false, select should be true.
+    ///   - width: The width of the image; it is set to default at 110.
+    ///   - height: The height of the image; it is set to default at 110.
+    func viewImage(
+        placeId: String,
+        localMarker: GMSMarker,
+        imageView: UIImageView,
+        select: Bool = false,
+        tapped: Bool = true,
+        width: Int = 110,
+        height: Int = 110
+    ) {
+        let placesClient: GMSPlacesClient = GMSPlacesClient.shared()
+        let fields: GMSPlaceField = .photos
+        placesClient.fetchPlace(fromPlaceID: placeId, placeFields: fields, sessionToken: nil, callback: {
+            (place: GMSPlace?, error: Error?) in
+            guard error == nil else {
+                print("Some error occured here: \(error?.localizedDescription ?? "")")
+                return
+            }
+            guard place != nil else {
+                print("The location is nil or does not exist: \(error?.localizedDescription ?? "")")
+                return
+            }
+            guard let place = place else {
+                print("Error loading photo metadata: \(error?.localizedDescription ?? "")")
+                return
+            }
+            if place.photos != nil {
+                guard let photoMetadata = place.photos?[0] else {
+                    print("There is no photo data for location: \(error?.localizedDescription ?? "")")
+                    return
+                }
+                placesClient.loadPlacePhoto(photoMetadata, callback: { (photo, error) -> Void in
+                    guard error == nil else {
+                        print("Some error occured: \(error?.localizedDescription ?? "")")
+                        return
+                    }
+                    if !select {
+                        let size = CGSize(width: width, height: height)
+                        UIGraphicsBeginImageContextWithOptions(size, false, 0.0);
+                        photo?.draw(in: CGRect(x: 0, y: 0, width: size.width, height: size.height))
+                        let newImage:UIImage = UIGraphicsGetImageFromCurrentImageContext()!
+                        UIGraphicsEndImageContext()
+                        let tempImage = newImage.opac(alpha: 0.7)
+                        localMarker.icon = tempImage?.circleMask
+                    } else {
+                        let size = CGSize(width: self.dim, height: self.dim)
+                        UIGraphicsBeginImageContextWithOptions(size, false, 0.0);
+                        photo?.draw(in: CGRect(x: 0, y: 0, width: size.width, height: size.height))
+                        let newImage: UIImage = UIGraphicsGetImageFromCurrentImageContext()!
+                        UIGraphicsEndImageContext()
```

Comment:
I think I have to still include a little part of the else statement since the card layout is different than the icon layout. This has definitely made the code shorter, though!